compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5326,6 +5311,55 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
         resultType = types.checkType(checkedExpr, actualType, expType);
     }
 
+    private void rewriteWithEnsureTypeFunc(BLangCheckedExpr checkedExpr, BType type) {
+        BType rhsType = getCandidateType(checkedExpr, type);
+        if (rhsType == symTable.semanticError) {
+            rhsType = checkExpr(checkedExpr.expr, env);
+        }
```

Comment:
Shouldn't we add an else block and call `checkExpr(checkedExpr.expr, env, type)`? We should type check the actual expr, since we type check a clone in `getCandidateType` right?

Reply:
that type checking logic is there in the line number 5238

Reply:
Ah okay. But then, we should use a clone in L5346 also, right?

Reply:
No right? If we do a clone, then original `checkExpr.expr` will be type checked on the normal path and it will report type incompatible error.

Reply:
ah yes. I got your point :)